fhirAssembly/build.gradle

```diff
@@ -0,0 +1,213 @@
+}
+
+configurations {
+    testResources.extendsFrom(testRuntimeOnly) // added so configurations search is allowed
+}
+
+dependencies {
+    testImplementation "org.slf4j:slf4j-api:1.7.25"
+    testImplementation "org.apache.logging.log4j:log4j-slf4j-impl:${log4jVersion}"
+    testImplementation "org.projectlombok:lombok:${lombokVersion}"
+    testImplementation 'com.fasterxml.jackson.core:jackson-databind:2.17.2'
+
+    // Used for uploading documents (images) to VANTIQ with VANTIQ SDK
+    testImplementation "io.vantiq:vantiq-sdk:${vantiqSDKVersion}"
+
+    // Use JUnit test framework
+    testImplementation 'junit:junit:4.13.1'
+    // Data to preload into our FHIR server
+    testResources "vantiq.testArtifacts:fhirR4Sample:1.0@zip"
+}
+
+tasks.register('unzipSampleData', Copy) {
+    outputs.upToDateWhen { new File(layout.buildDirectory.dir('sampleData').get().toString()).exists()}
+    def zipPath = project.configurations.testResources.find {
+        it.name.startsWith("fhirR4Sample")
+    }
+    logger.debug('Unzipping data from {}', zipPath)
+    def zipFile = file(zipPath)
+
+    from zipTree(zipFile)
+    into layout.buildDirectory.dir('sampleData')
+}
+
+/**
+ * We start our server empty so we'll load a few entries from the synthea sample set.  We aren't testing the FHIR
+ * server, just our ability to access it, so
+ */
+tasks.register('uploadSampleData') {
+    inputs.dir(layout.buildDirectory.dir('sampleData').get())
+    outputs.upToDateWhen { false }
+    doLast {
+        File dir = new File(layout.buildDirectory.dir('sampleData').get().toString())
+        String fhirServer = "${fhirServerUrl}"
+        int loaded = 0
+        int errors = 0
+        int redirects = 0
+        int attempts = 0
+        dir.listFiles().each { sampleFile ->
+            def redirectedServer = fhirServer
+            try {
+                attempts += 1
+                def redirecting = true
+                while (redirecting) {
+                    HttpURLConnection req = new URL(redirectedServer).openConnection() as HttpURLConnection
+                    req.setFixedLengthStreamingMode(sampleFile.length())
+                    req.setRequestMethod('POST')
+                    req.setRequestProperty('Content-Type', 'application/json; charset=UTF-8')
+                    // Note:  If accept header isn't set (or not set to app/json or app/xml), you get HTML back,
+                    // and the work isn't actually done.  Only took a day to figure that out :-(
+                    req.setRequestProperty('Accept', 'application/json')
+                    req.setDoOutput(true)
+                    req.setDoInput(true)
+                    req.getOutputStream().write(sampleFile.bytes)
+                    logger.debug('Uploading to {}: {}', redirectedServer, new String(sampleFile.bytes))
+                    logger.info "Status code: ${req.getResponseCode()}" // HTTP request done on first read
+                    switch (req.getResponseCode()) {
```

Comment:
Have to handle redirects here due to the `setFixedLengthStreamingMode()` call.  At least one of the samples is long enough that it needs to be chunked, so...